src/sentry/notifications/notificationcontroller.py

```diff
@@ -307,8 +317,9 @@ def _get_layered_setting_providers(
                 for provider_str in PERSONAL_NOTIFICATION_PROVIDERS:
                     provider = ExternalProviderEnum(provider_str)
                     if provider not in most_specific_recipient_providers[type]:
-                        # TODO: fix so team defaults don't have different logic
-                        if recipient_is_team(recipient):
+                        if recipient_is_team(recipient) and (
+                            (not has_team_workflow) or (provider not in TEAM_NOTIFICATION_PROVIDERS)
```

Comment:
I was hoping we could remove this check instead of making it more complicated :(

Reply:
Yeah I'm trying to start to remove it in a separate [PR](https://github.com/getsentry/sentry/pull/58972) just so that it doesn't mess up any of the team issue alert notification settings. We can remove the check entirely once the feature isn't behind a flag